policy/manager.go

```diff
@@ -149,3 +180,23 @@ func (m *Manager) EnforceCooldown(id string, t time.Duration) {
 		m.log.Debug("attempted to set cooldown on non-existent handler", "policy_id", id)
 	}
 }
+
+// isUnrecoverableError checks if the input error should be considered
+// unrecoverable.
+//
+// An error is considered unrecoverable if it has the potential to affect the
+// policy manager's internal state, and so the policy manager should re-run.
+//
+// Example of an unrecoverable error: the Nomad server becomes unreachable;
+// upon reconnecting, the state of the Nomad cluster could be different from
+// what's stored in the policy manager (e.g., a different cluster becomes
+// available in the same address).
+func isUnrecoverableError(err error) bool {
+	unrecoverableErrors := []string{"connection refused", "EOF"}
```

Comment:
The net stdlib package has [custom errors](https://github.com/golang/go/blob/4ec4a792f6e60c9c4f40a928650c66419ee0e446/src/net/net.go#L392) which have a `Temporary()` function. I am not sure this would cover the cases you have specifically found, but something worth noting at least.

Reply:
TIL

But the problem is when the error reaches the policy manager it has been transformed into a generic `error`. We could start considering using [`error.Wrap`](https://godoc.org/github.com/pkg/errors#Wrap) instead of creating new error objects